### PR TITLE
Backports 0.19 pr2

### PR DESCRIPTION
**Pull request by Munkybooty**

### CODE REVIEW

**Comment by Munkybooty** [comment]
not sure how to get #15896 to not fail the tests

File: test/functional/feature_filelock.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def run_test(self):
         self.log.info("Using datadir {}".format(datadir))
 
         self.log.info("Check that we can't start a second dashd instance using the same datadir")
-        expected_msg = "Error: Cannot obtain a lock on data directory {}. Dash Core is probably already running.".format(datadir)
+        expected_msg = "Error: Cannot obtain a lock on data directory {0}. {1} is probably already running.".format(datadir, self.config['environment']['PACKAGE_NAME'])
```
Comment (UdjinM6) [inline test/functional/feature_filelock.py:26]: 15896: requires 13926 or at least https://github.com/UdjinM6/dash/commit/1638efb830fe460affd6c8e9d8e347bc3c90165a

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by Munkybooty** [comment]
I dropped 15788 and 15843 as asked, I'm hoping i got everything when going for dashification

**Comment by Munkybooty** [comment]
Totally thought it was already merged, but i'll remember that!

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see https://github.com/UdjinM6/dash/commits/pr4481

**Comment by UdjinM6** [comment]
pls see 57d6c49997

File: src/bitcoin-wallet.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    gArgs.AddArg("-wallet=<wallet-name>", "Specify wallet name", false, OptionsCategory::OPTIONS);
+    gArgs.AddArg("-debug=<category>", "Output debugging information (default: 0).", false, OptionsCategory::DEBUG_TEST);
+    gArgs.AddArg("-printtoconsole", "Send trace/debug info to console (default: 1 when no -debug is true, 0 otherwise.", false, OptionsCategory::DEBUG_TEST);
+
+    gArgs.AddArg("info", "Get wallet info", false, OptionsCategory::COMMANDS);
+    gArgs.AddArg("create", "Create new wallet file", false, OptionsCategory::COMMANDS);
+
+    // Hidden
+    gArgs.AddArg("-h", "", false, OptionsCategory::HIDDEN);
+    gArgs.AddArg("-help", "", false, OptionsCategory::HIDDEN);
```
Comment (UdjinM6) [inline src/bitcoin-wallet.cpp]: 13926: this will have to be updated with s/`false`/`ArgsManager::ALLOW_ANY`/ after #4569

Comment (Munkybooty) [reply src/bitcoin-wallet.cpp]: I have the changes committed and will push them as soon as #4569 is merged

**Comment by PastaPastaPasta** [comment]
Can we at least squash the fixes? Or maybe squash the fixes into the original

**Comment by Munkybooty** [comment]
They are squashed on my local branch so as soon as we merge #4569 i'll push the squashed one with the additional changes Udjin requested

**Comment by UdjinM6** [comment]
Looks good 👍 pls squash 3876682 into 13926

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**